Beam - fixed inconsistencies of draw method

Davide-sd opened the pull request:
<!-- Your title above should be a short description of what
was changed. Do not include the issue number in the title. -->

#### References to other Issues or PRs
<!-- If this pull request fixes an issue, write "Fixes #NNNN" in that exact
format, e.g. "Fixes #1234" (see
https://tinyurl.com/auto-closing for more information). Also, please
write a comment on that issue linking back to this pull request once it is
open. -->

Fixes #24043 


#### Brief description of what is fixed or changed

This PR is fixing the inconsistencies mentioned on Issue #24043 , thus providing a schematic view which is consistent with the sign convention used by the `Beam` class. Not only this improves the visualization. It also allows the users to better make sense of the symbolic results. In particular:

* the `draw` method is no longer looking at the assumptions on symbolic loads and moments, which are not used by the `Beam` class either in its computations.
* positive loads (numeric or symbolic) are pointing downward.
* negative loads (numeric or symbolic) are pointing upward.
* positive moments (numeric or symbolic) are directed counterclockwise.
* negative moments (numeric or symbolic) are directed clockwise.
* the `draw` method is now able to visualize symbolic distributed loads without raising errors.
* the beam won't be hidden by the distributed loads.


#### Other comments


#### Release Notes

<!-- Write the release notes for this release below between the BEGIN and END
statements. The basic format is a bulleted list with the name of the subpackage
and the release note for this PR. For example:

* solvers
  * Added a new solver for logarithmic equations.

* functions
  * Fixed a bug with log of integers.

or if no release note(s) should be included use:

NO ENTRY

See https://github.com/sympy/sympy/wiki/Writing-Release-Notes for more
information on how to write release notes. The bot will check your release
notes automatically to see if they are formatted correctly. -->

<!-- BEGIN RELEASE NOTES -->

* physics.continuum_mechanics
  * Improved the schematic view produced by the `Beam.draw` method.

<!-- END RELEASE NOTES -->

Davide-sd committed:
fixed draw method

moorepants commented:
@AdvaitPote, given that you are also working on draw methods for the trusses, it would be in your best interest to review this PR and discuss.

moorepants commented on sympy/physics/continuum_mechanics/beam.py:
```diff
@@ -2189,13 +2214,64 @@ def draw(self, pictorial=True):
+
+        Notes
+        =====
+        The Beam class doesn't care about assumptions on symbolic loads.
```
The Beam class *should* care about assumptions. That is the only reliable way to know if a symbol or expression involving symbols is positive or negative.

smichr replied:
It now cares and also tries doit and expand for non-atomic expressions if the give None in their raw state.

oscarbenjamin commented:
This has merge conflicts now. I'll mark this as draft. Feel free to mark it as ready for review later.

smichr committed:
Merge branch 'master' into issue_24043

smichr commented on sympy/physics/continuum_mechanics/beam.py:
```diff
@@ -2213,95 +2299,88 @@ def _draw_load(self, pictorial, length, l):
+                    warning_body += "* Point load %s located at %s\n" % (load[0], load[1])
+
+                if iln:
+                    annotations.append({'text':'', 'xy':(pos, 0), 'xytext':(pos, height - 4*height), 'arrowprops':dict(width= 1.5, headlength=5, headwidth=5, facecolor='black')})
```
```suggestion
                    annotations.append({'text':'', 'xy':(pos, 0), 'xytext':(pos, height - 4*height), 'arrowprops':{'width': 1.5, 'headlength': 5, 'headwidth': 5, 'facecolor': 'black'}})
```

smichr committed:
remove dict(...) use

I wish that were not flagged: it is  convenient way to avoid having to type string quotation marks.

smichr committed:
use is_negative for _is_load_negative

smichr committed:
Update beam.py

smichr commented:
@AdvaitPote @moorepants , please give this another look.

smichr commented:
Further modifications can be made in a new PR.
